### DISCUSSION TITLE
[2023-03-06] Component Browser Team Update

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#5824] [Weekly Demo Day]
* Improving Component Browser, so it would be more cursor-position aware is still in progress. Still fixing bugs, but also improved UX by making special entry for putting literals. Once merged, nobody would need to remember pressing `cmd+enter` instead of `enter` when writing expressions like `2 + 2`!
* Fixed all known issues with lazy table visualization, the long-awaited PR is being reviewed.
* Investigated the problem with timeouts on cloud. Full report [here](https://github.com/enso-org/enso/issues/5627#issuecomment-1449971942), also implemented a quickfix: now GUI is retrying initialization several times, so we can wait out the cloud "warm-up".

Fixes:
* Created a fix for re-appeared duplicated entries issue
* Fixed the occasionally disappearing closing `)` when using the drop downs.
* Fixing the Component Browser icons not visible on some Windowses. Until merged, the workaround is to set screen scale to an integral value.

**Comment by farmaazon** [comment]
The screencast of initialization retrying. Only after three retries we display error message.

https://user-images.githubusercontent.com/3919101/223145451-a4128558-23d2-481a-bcbc-c858cf1ba791.mp4

@vitvakatu @MichaelMauderer please put screencasts of your demos here.

**Comment by vitvakatu** [comment]
Cursor aware behavior of the component browser:


https://user-images.githubusercontent.com/6566674/223209377-8d359cf6-fa65-414b-a20b-8e25f05f96a9.mp4


https://user-images.githubusercontent.com/6566674/223209417-132a1c10-12b0-42a4-bd59-d15ddbde3988.mp4

**Comment by MichaelMauderer** [comment]
![image](https://user-images.githubusercontent.com/1428930/223535048-a8f167f6-1264-4bde-94a5-f9711e214781.png)

https://user-images.githubusercontent.com/1428930/223535262-4714edd3-a876-4e73-86a8-54c16ea2763e.mp4
